src/NuGet.Clients/NuGet.CommandLine/Program.cs

```diff
@@ -227,16 +226,55 @@ private void Initialize(CoreV2.NuGet.IFileSystem fileSystem, IConsole console)
         // This method acts as a binding redirect
         private Assembly CurrentDomain_AssemblyResolve(object sender, ResolveEventArgs args)
         {
-            var name = new AssemblyName(args.Name);
+            AssemblyName name = new AssemblyName(args.Name);
 
             if (string.Equals(name.Name, ThisExecutableName, StringComparison.OrdinalIgnoreCase))
             {
                 return typeof(Program).Assembly;
             }
+            // .NET Framework 4.x now triggers AssemblyResolve event for resource assemblies
+            // Catch this event for nuget.exe (NuGet.CommandLine) and NuGet.Command resource assemblies only
+            if (name.Name == "NuGet.resources" || name.Name == "NuGet.Commands.resources")
+            {
+                // Load satellite resource assembly from embedded resources
+                return GetNuGetResourcesAssembly(name.Name, name.CultureInfo);
+            }
 
             return null;
         }
 
+        private static Assembly GetNuGetResourcesAssembly(string name, CultureInfo culture)
+        {
+            string resourceName = $"NuGet.CommandLine.{culture.Name}.{name}.dll";
+            Assembly resourceAssembly = LoadAssemblyFromEmbeddedResources(resourceName);
+            if (resourceAssembly == null)
+            {
+                // Sometimes, embedded assembly names have dashes replaced by underscores
+                string altResourceName = $"NuGet.CommandLine.{culture.Name.Replace("-", "_")}.{name}.dll";
+                resourceAssembly = LoadAssemblyFromEmbeddedResources(altResourceName);
+            }
+
+            return resourceAssembly;
+        }
+
+        private static Assembly LoadAssemblyFromEmbeddedResources(string resourceName)
+        {
+            Assembly resourceAssembly = null;
+            using (var stream = Assembly.GetExecutingAssembly().GetManifestResourceStream(resourceName))
```

Comment:
You could extract `Assembly.GetExecutingAssembly()` into a static readonly field, so it doesn't need to be "calculated" each time. Also, I have a personal preference of `typeof(Program).Assembly` over `GetExecutingAssembly`.

Reply:
It sounds good to cache Assembly.GetExecutingAssembly(). I will try and update the PR as needed.

Reply:
Changed for this class. Other parts may be using `Assembly.GetExecutingAssembly()` but, I will not change those, since those operations made more sophisticated stuff.